backend/http/http.go

```diff
@@ -653,6 +652,12 @@ func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (in io.Read
 	if err != nil {
 		return nil, fmt.Errorf("Open failed: %w", err)
 	}
+
+	if o.fs.opt.NoHead {
```

Comment:
We should probably do this regardless of `o.fs.opt.NoHead` - that is what the other backends do - the response  fields should be accurate after a `GET` request.

Reply:
Make sense.  I can't figure out any scenario where `HEAD` returns the correct modTime while `GET` do not.